### PR TITLE
Allow skipping sell/undeploy anim

### LINKED ISSUE #13688: TransformOnCapture plays sell animations
The new transform activity code plays sell animation all the time, even when SkipMakeAnims == true (by default). Considering sell and make are different things I think there should be SkipSellAnims option as well.

### LINKED ISSUE #13704: WithMakeAnimation interferes structures with GrantConditionOnDeploy
I have a Radar Dome that upon deploy, gets into high power mode and consumes 400 power granting sight range of 30 cells. From the new play test, the deploy now triggers sell/make animations.

This gets in the way of RA2 Super Gap Generators too: deploy to make it consume a lot of power for making wider gap.

Make animation Needs to be conditional or must be filtered by deploy type?
Or better, "Building Complete" (this is what the flag is called in building.cs) should grant a condition.
I have a structure that fires explosion all the time but, it looks awkward when the weapon fires while the build animation is playing.

### PR DESCRIPTION
**Pull request by reaperrr**
At least in `TransformOnCapture` and `Sellable`.

Closes #13688.
Closes #13704.

### CODE REVIEW

File: OpenRA.Mods.Common/Activities/Transform.cs [hunk 1]
```diff
@@ -27,6 +27,7 @@ public class Transform : Activity
 		public string Notification = null;
 		public int ForceHealthPercentage = 0;
 		public bool SkipMakeAnims = false;
+		public bool SkipReversedMakeAnim = false;
```
Comment (pchote) [inline OpenRA.Mods.Common/Activities/Transform.cs]: Please pick one of "Anim" or "Anims" for both

File: OpenRA.Mods.Common/Traits/Sellable.cs [hunk 2]
```diff
@@ -23,6 +23,9 @@ public class SellableInfo : ConditionalTraitInfo
 		public readonly int RefundPercent = 50;
 		public readonly string[] SellSounds = { };
 
+		[Desc("Skip sell animations.")]
+		public readonly bool SkipSellAnim = false;
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Sellable.cs]: Can we please call this `SkipMakeAnimation`?

File: OpenRA.Mods.Common/Traits/TransformOnCapture.cs [hunk 3]
```diff
@@ -20,6 +20,9 @@ public class TransformOnCaptureInfo : ITraitInfo
 		public readonly int ForceHealthPercentage = 0;
 		public readonly bool SkipMakeAnims = true;
 
+		[Desc("Skip undeploy/sell animation.")]
+		public readonly bool SkipReversedMakeAnim = true;
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/TransformOnCapture.cs]: Likewise here?

Comment (pchote) [inline OpenRA.Mods.Common/Traits/TransformOnCapture.cs]: Do we have an actual use case for separating forward vs reverse animations?

**Review by obrakmann — APPROVED** [review]
lgtm, 👍

File: OpenRA.Mods.Common/Traits/TransformOnCapture.cs (hunk 3 above)
Comment (reaperrr) [inline OpenRA.Mods.Common/Traits/TransformOnCapture.cs]: Not in our shipping mods. I guess I'll just use `SkipMakeAnims` for now.

**Comment by reaperrr** [comment]
Updated.

File: OpenRA.Mods.Common/Traits/Sellable.cs [hunk 4]
```diff
@@ -63,9 +66,14 @@ public void Sell(Actor self)
...
-			var makeAnimation = self.TraitOrDefault<WithMakeAnimation>();
-			if (makeAnimation != null)
-				makeAnimation.Reverse(self, new Sell(self), false);
+			if (!info.SkipMakeAnimation)
+			{
+				var makeAnimation = self.TraitOrDefault<WithMakeAnimation>();
+				if (makeAnimation != null)
+					makeAnimation.Reverse(self, new Sell(self), false);
+				else
+					self.QueueActivity(false, new Sell(self));
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Sellable.cs]: Duplicating this `QueueActivity` isn't great.  If you return after the `makeAnimation.Reverse` then you can have just one at the end of the method instead.

**Comment by reaperrr** [comment]
Updated.

**Comment by MustaphaTR** [comment]
While you are in it, can we get a SkipMakeAnimation for GrantConditionOnDeploy. That could fix #13704, i'm having similar problem with Chinese Nuke Reactors (Overcharge) too.

**Comment by pchote** [comment]
@MustaphaTR: Good point.  IMO that is a more serious problem than the sell case because modders are more likely to use `GrantConditionOnDeploy` than `TransformOnCapture`.

**Comment by reaperrr** [comment]
Updated.

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/e89d42b3d0e70990875232fc49eb7bb3ad1479d8

4ca87c1e05ccfa1bdcf08031e369e834e83bf0fa
619d433e8bbd3aec9b45a334dc3edde28b5495ae
740f04801b40538ca3f18c036e22b359a91f43f3

**Pull request merged**